### PR TITLE
[OSDEV-1373] SLC. Implement search page for name & address search (FE).

### PR DESCRIPTION
**Pull request by Innavin369**
[[OSDEV-1373] ](https://opensupplyhub.atlassian.net/browse/OSDEV-1373) The tab `Search by Name and Address.` on the Production Location Search screen has been implemented. There are three required properties (name, address, country). The "Search" button becomes clickable after filling out inputs, creates a link with parameters, and allows users to proceed to the results screen.

Screenshot 2024-12-02 at 15 37 57

[OSDEV-1373]: https://opensupplyhub.atlassian.net/browse/OSDEV-1373?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/react/src/components/Contribute/SearchByNameAndAddressTab.jsx [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+    const handleSearch = () => {
+        // history.push(
+        //     `/contribute/production-location/search/?name=${inputName}&address=${inputAddress}&country=${inputCountry.value}`,
+        // );
+    };
+    const isDisabled =
+        validate(inputName) &&
+        validate(inputAddress) &&
+        validate(inputCountry.label);
```
[the remark this answers is not in the document]
Comment (Innavin369) [reply src/react/src/components/Contribute/SearchByNameAndAddressTab.jsx:67]: changed

File: src/react/src/components/Filters/StyledSelect.jsx [hunk 2]
```diff
@@ -28,13 +28,15 @@ function StyledSelect({
...
-            </InputLabel>
+            {label && (
+                <InputLabel
+                    shrink={false}
+                    htmlFor={name}
+                    className={classes.inputLabelStyle}
+                >
+                    {label} {renderIcon()}
+                </InputLabel>
+            )}
```
[the remark this answers is not in the document]
Comment (Innavin369) [reply src/react/src/components/Filters/StyledSelect.jsx:39]: The label creates an element in the DOM  and if there is no value I have additional padding (I don't need it) and a warning in the browser console. So I decided to make the label optional.

**Review by VadimKovalenkoSNF — CHANGES_REQUESTED** [review]
Request minor changes.

File: src/react/src/util/COLOURS.js [hunk 3]
```diff
@@ -11,6 +11,7 @@ export default {
 
     // Reds
     LIGHT_RED: '#FFEAEA',
+    RED: '#f44336',
```
[the remark this answers is not in the document]
Comment (Innavin369) [reply src/react/src/util/COLOURS.js:14]: done

File: src/react/src/components/Contribute/SearchByNameAndAddressTab.jsx [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+
+    const handleSearch = () => {
+        const baseUrl = '/contribute/production-location/search/';
+        const url = `${baseUrl}?name=${encodeURIComponent(
+            inputName,
+        )}&address=${encodeURIComponent(
+            inputAddress,
+        )}&country=${encodeURIComponent(inputCountry.value ?? '')}`;
+        history.push(url);
+    };
```
[the remark this answers is not in the document]
Comment (Innavin369) [reply src/react/src/components/Contribute/SearchByNameAndAddressTab.jsx:75]: changed

File: src/react/src/__tests__/components/ContributeProductionLocation.test.js [hunk 5]
```diff
@@ -34,23 +34,23 @@ describe('ContributeProductionLocation component', () => {
...
-        fireEvent.click(nameAddressTab);
+        const osIdTab = getByRole('tab', { name: /Search by OS ID/i });
+        fireEvent.click(osIdTab);
 
-        expect(nameAddressTab).toHaveAttribute('aria-selected', 'true');
-        expect(getByRole('tab', { name: /Search by OS ID/i })).toHaveAttribute('aria-selected', 'false');
-        expect(getByText('Search by Name and Address Tab')).toBeInTheDocument();
+        expect(osIdTab).toHaveAttribute('aria-selected', 'true');
+        expect(getByRole('tab', { name: /Search by Name and Address/i })).toHaveAttribute('aria-selected', 'false');
+        expect(getByText('Search by OS ID')).toBeInTheDocument();
```
[the remark this answers is not in the document]
Comment (VadimKovalenkoSNF) [reply src/react/src/__tests__/components/ContributeProductionLocation.test.js:42]: @Innavin369 , could you apply this test case?

[the remark this answers is not in the document]
Comment (VadimKovalenkoSNF) [reply src/react/src/__tests__/components/ContributeProductionLocation.test.js:42]: Ok, you can skip tests for Select component.

File: src/react/src/components/Contribute/SearchByNameAndAddressTab.jsx [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+    const isFormValid =
+        validate(inputName) &&
+        validate(inputAddress) &&
+        validate(inputCountry.value);
+
+    useEffect(() => {
+        if (!countriesData) {
+            fetchCountries();
+        }
+    }, []);
```
[the remark this answers is not in the document]
Comment (Innavin369) [reply src/react/src/components/Contribute/SearchByNameAndAddressTab.jsx]: added dependencies, skipped the rest changes

File: src/react/src/components/Contribute/SearchByNameAndAddressTab.jsx [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+            alpha_3: 'GBR',
+            numeric: '826',
+        },
+        {
+            name: 'United States of America',
+            alpha_2: 'US',
+            alpha_3: 'USA',
+            numeric: '840',
+        },
+    ];
```
[the remark this answers is not in the document]
Comment (Innavin369) [reply src/react/src/components/Contribute/SearchByNameAndAddressTab.jsx]: done

File: src/react/src/components/Contribute/SearchByNameAndAddressTab.jsx [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+    };
+    const handleCountryChange = event => {
+        setInputCountry(event || '');
+    };
+
+    const handleSearch = () => {
+        // history.push(
+        //     `/contribute/production-location/search/?name=${inputName}&address=${inputAddress}&country=${inputCountry.value}`,
+        // );
+    };
```
[the remark this answers is not in the document]
Comment (Innavin369) [reply src/react/src/components/Contribute/SearchByNameAndAddressTab.jsx]: changed

**Review by VadimKovalenkoSNF — APPROVED** [review]
Approved.

**Review by roman-stolar — APPROVED** [review]
LGTM

**Pull request merged**